packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -799,53 +804,74 @@ export const sendGsheetsNotification = async (
             targetType: 'gsheets',
             status: SchedulerJobStatus.STARTED,
         });
-
-        const { url, pageType, csvUrl, csvUrls } = notification.page;
+        const user = await userService.getSessionByUserUuid(
+            scheduler.createdBy,
+        );
 
         if (format !== SchedulerFormat.GSHEETS) {
             throw new Error(
                 `Unable to process format ${format} on sendGdriveNotification`,
             );
         } else if (savedChartUuid) {
-            if (csvUrl === undefined) {
-                throw new Error('Missing CSV URL');
-            }
+            const rows = await projectService.getResultsForChart(
+                user,
+                savedChartUuid,
+            );
 
-            const csvContent = await csv().fromFile(csvUrl.localPath);
             // TODO use csv().fromStream to do stream reading/writting on gsheets
             const refreshToken = await userService.getRefreshToken(
                 scheduler.createdBy,
             );
-            await googleDriveClient.appendToSheet(
-                refreshToken,
-                gdriveId,
-                csvContent,
-            );
+            await googleDriveClient.appendToSheet(refreshToken, gdriveId, rows);
```

Comment:
The arguments and types for `googleDriveClient` don't look right. It's expecting csvContent but now it gets rows. Is it writing the header correctly?

Reply:
csvContent was Record<string,string>[] which is the same type as rows. Afaik the first row contains the headers. Note. this is before we `format rows`

Reply:
rows are `Record<string,any>` so it's not erroring, really rows should be `Record<string,unknown>` then you'd have seen a compile error here